net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/common/OpenVpn.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/*
+ * Copyright (C) 2018 Eugen Mayer
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Freeradius\common;
+// yeah, why should plugins.inc.d/openvpn.inc include all the symbols it is using..
+require_once("util.inc");
+require_once("plugins.inc.d/openvpn.inc");
+
+use \OPNsense\Core\Config;
+
+/**
+ * Handles all kind of OpenVPN based operations
+ * Class OpenVpn
+ * @package OPNsense\Freeradius\common
+ */
+class OpenVpn
+{
+
+    /**
+     * @param CCD $dynamicCDD
+     * @param null $servers
+     */
+    static public function resetToStaticOrDelete(CCD $dynamicCDD, $servers = null)
+    {
+        if ($servers == NULL) {
+            $servers = self::getDynamicCCDopenVPNServers();
+        }
+        $staticCCDs = self::getOpenVpnCCD();
+        if (isset($staticCCDs[$dynamicCDD->common_name])) {
+            $ccd = $staticCCDs[$dynamicCDD->common_name];
+            foreach ($servers as $server) {
+                // thats a openvpn legacy tool to create CCDs for a specific server
+                // lets use this to ensure compatibility
+                $ccdConfigAsString = openvpn_csc_conf(self::ccdToLegacyStructure($ccd), $server);
+                // this will override - reset the overlayed one with the original from static
+                self::writeCCDforServer($ccd->common_name, $ccdConfigAsString, $server['vpnid']);
+            }
+        } else {
+            // since the CCD does not exist in static, remove it. This is already more then the current core implementation does
+            // it does nothing in this case
```

Comment:
move some words from 64 to 65 to shorten the line. Would look far better.

Reply:
done